File: 01/go/part_two.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"sort"
	"strconv"
)

func main() {
	scanner := bufio.NewScanner(os.Stdin)

	counts := make([]int, 1000)

	elves := 0
	for scanner.Scan() {
		line := scanner.Text()
		if line == "" {
			elves++
		} else {
			num, _ := strconv.Atoi(line)
			counts[elves] += num
		}
	}

	sort.Sort(sort.Reverse(sort.IntSlice(counts)))

	sum_top_three := counts[0] + counts[1] + counts[2]

	fmt.Println(sum_top_three)
}

File: 01/python/part_two.py
import sys

elves = 0
calories = [0 for i in range(1000)]

for line in sys.stdin:
    line = line.rstrip()

    if line == "":
        elves += 1
    else:
        calories[elves] += int(line)

calories.sort(reverse=True)
top_three = calories[0:3]
sum_top_three = sum(top_three)

print(sum_top_three)
